mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -259,10 +271,12 @@ def evaluate_if(self, node: mparser.IfClauseNode) -> T.Optional[Disabler]:
             # statement evaluation.
             self.tmp_meson_version = None
             result = self.evaluate_statement(i.condition)
+            if result is None:
+                raise InvalidCodeOnVoid()
```

Comment:
We would catch this as a MesonBug a couple lines below, but raising it as a bug is indeed awkward. :)